vault_cli/client.py

```diff
@@ -217,6 +237,16 @@ def _merge_secrets(self, secrets: types.JSONDict) -> types.JSONDict:
 
         return secrets
 
+    def delete_all_secrets(self, *paths: str) -> Iterable[str]:
+        """
+        Recursively deletes all the secrets at the given paths.
+        """
+        for path in paths:
+            secrets_paths = self._browse_recursive_secrets(path=path)
+            for secret_path in secrets_paths:
+                yield secret_path
```

Comment:
it's weird that this method `yield` the item before deleting item :)

Reply:
Yes, to give a chance for confirming your choice before it’s too late. Is it that much of a strange pattern ?

Reply:
the goal is clear but the name of the function is misleading :)

Reply:
Any suggestion ?

Reply:
`yield_and_delete_all_secrets` :D

Reply:
`pop_all_secrets` ? :)

Reply:
Je vais merger et on améliorera ça après